### PR TITLE
Changing way to use original data within histogram in order to fix 'vector histograms'

### LINKED ISSUE #195: finish time slider for static layers
it does not have for example "clear selection" and so on

### LINKED ISSUE #277: When a torque widget is filtered, widget info is rendered twice
It is related how views are listening changes to renderRange attribute, we should get rid of it because filter model already have the necessary info.

### PR DESCRIPTION
**Pull request by xavijam**
Basically, after [this change](https://github.com/CartoDB/cartodb.js/pull/1181) in CartoDB.js, there is a new way to obtain all the data of the widget. That means we need to change several things in the histogram `content-view` and `chart` classes. I've tried to keep things as they are and only touch affected parts. I'd like to know your opinion @javierarce before keep going.

cc @viddo @javisantana 

Pending things:
- [x] Review tests.
- [x] Add new tests.
- [x] Review the impact over time series/torque widget.
- [x] Review static time-slider widget (fixes #195).
- [x] Review torque info when it is selected (fixes #277).

Overall, fixes https://github.com/CartoDB/cartodb.js/issues/1179.

CR: @javierarce

### CODE REVIEW

File: src/widgets/histogram/chart.js [hunk 1]
```diff
@@ -497,15 +493,22 @@ module.exports = cdb.core.View.extend({
...
+  _getDataForScales: function () {
+    if (this.model.get('bounded')) {
+      return this.model.get('data');
+    } else {
+      return this._originalData && this._originalData.getData();
+    }
+  },
+
   _setupScales: function () {
+    var data = this._getDataForScales();
```
Comment (xavijam) [inline src/widgets/histogram/chart.js]: Scale of the histogram depends on the state of itself, if it is bounded (/zoomed) or not.

File: src/widgets/histogram/content-view.js [hunk 2]
```diff
@@ -289,21 +277,12 @@ module.exports = cdb.core.View.extend({
...
-  },
-
   _onChangeFilterEnabled: function () {
     this.$('.js-filter').toggleClass('is-hidden', !this.model.get('filter_enabled'));
   },
 
-  _onChangeBins: function () {
-    this._originalData.reset([]); // Clean originalData
+  _onChangeBins: function (mdl, bins) {
+    this._originalData.setBins(bins);
```
Comment (xavijam) [inline src/widgets/histogram/content-view.js]: This will trigger a new request for getting all the data with that bins.

**Comment by viddo** [comment]
Make sure to verify torque/normal time-series widget still working too, since they also depend on the chart. Just by looking at the code I think there should not be any trouble, but better safe than sorry.

**Comment by xavijam** [comment]
Ha! I didn't finish the pending things in the description, but it was the second point:

screen shot 2016-03-10 at 11 16 05

:scream: 

Thanks @viddo :heart:, now description updated.

**Comment by xavijam** [comment]
About last point in the pending list, self-explanatory gifs:

Before:
torque-bug

Now:
torque-fix

I'd like you @javisantana @viddo  to check the solution, in any case I'd like to know more about the `renderRange` later @javisantana.

**Comment by xavijam** [comment]
So guys, if tests don't play with me, I have this thing ready to be reviewed.
In short:

- Histograms reviewed, the main change using a new model to get all the data.
- Time series widget checked, now it displays "clear" button and selected dates when it is filtered.
- Torque widget "fixed", there was a problem with CartoDB.js and the dates when it was filtered.

@javierarce,  your turn :heart:.

**Comment by javierarce** [comment]
:+1: LGTM

**Comment by xavijam** [comment]
:dancer: :dancers: :dancer: :dancers:

**Pull request merged**